packages/typespec-ts/src/transform/transform.ts

```diff
@@ -103,14 +107,76 @@ export async function transformRLCModel(
     apiVersionInfo,
     parameters,
     helperDetails,
-    urlInfo
+    urlInfo,
+    telemetryOptions
   };
 }
 
-export function transformUrlInfo(
-  program: Program,
-  dpgContext: SdkContext
-): UrlInfo | undefined {
+function transformTelemetryOptions(
+  dpgContext: SdkContext,
+  client: SdkClient
+): TelemetryOptions | undefined {
+  const customRequestIdHeaderName = getCustomRequestHeaderNameForClient(
+    dpgContext,
+    client
+  );
+  if (customRequestIdHeaderName) {
+    return {
+      customRequestIdHeaderName
+    };
+  }
+  return undefined;
+}
+
+function getCustomRequestHeaderNameForClient(
+  dpgContext: SdkContext,
+  client: SdkClient
+) {
+  const program = dpgContext.program;
+  const operationGroups = listOperationGroups(dpgContext, client);
+  for (const operationGroup of operationGroups) {
+    const operations = listOperationsInOperationGroup(
+      dpgContext,
+      operationGroup
+    );
+    for (const op of operations) {
+      const headerName = getCustomRequestHeaderNameForOperation(
+        ignoreDiagnostics(getHttpOperation(program, op))
+      );
+      if (headerName != undefined) {
+        return headerName;
+      }
+    }
+  }
+  const clientOperations = listOperationsInOperationGroup(dpgContext, client);
+  for (const clientOp of clientOperations) {
+    const headerName = getCustomRequestHeaderNameForOperation(
+      ignoreDiagnostics(getHttpOperation(program, clientOp))
+    );
+    if (headerName != undefined) {
+      return headerName;
+    }
+  }
+  return undefined;
+}
+
+function getCustomRequestHeaderNameForOperation(
+  route: HttpOperation
+): string | undefined {
+  const CUSTOM_REQUEST_HEADER_NAME = "client-request-id";
+  const params = route.parameters.parameters.filter(
+    (p) =>
+      p.type === "header" && p.name.toLowerCase() === CUSTOM_REQUEST_HEADER_NAME
+  );
+  if (params.length > 0) {
+    return CUSTOM_REQUEST_HEADER_NAME;
```

Comment:
feel a liitle weird that an operation level header parameter need to be implemented in the client level.

What if there're some operations has this header and some operation doesn't have this header ?

Reply:
This is current agreement cross languages, and I think JS could also follow this.